src/esp/gfx/CMakeLists.txt

```diff
@@ -1,14 +1,30 @@
-file(GLOB_RECURSE gfx_SOURCES "*.cpp")
+set(gfx_SOURCES
+  Drawable.cpp
+  GenericDrawable.cpp
+  GenericShader.cpp
+  RenderCamera.cpp
+  Renderer.cpp
+)
+
+# If not WebGL build, also include offscreen render-reliant code
+# TODO: this should be built as well (with emscripten windowless contexts)
+if(NOT BUILD_WEBGL)
+  list(APPEND gfx_SOURCES Simulator.cpp WindowlessContext.cpp)
```

Comment:
The Magnum builtin `WindowlessEglApplication` works as-is on Emscripten, but I'm not sure about your modified version -- the extension APIs most definitely aren't defined there. I can take a look at this once I get to integrating your WindowlessContext modifications back into Magnum.